src/probability/martingale/convergence.lean

```diff
@@ -231,4 +251,242 @@ mem_ℒp_limit_process_of_snorm_bdd
 
 end ae_convergence
 
+section L1_convergence
+
+variables [is_finite_measure μ] {g : Ω → ℝ}
+
+/-!
+
+### L¹ martingale convergence theorem
+
+We will now prove the L¹ martingale convergence theorems.
+
+The L¹ martingale convergence theorem states that:
+(a) if `f` is a uniformly integrable (in the probability sense) submartingale adapted to the
+  filtration `ℱ`, it converges in L¹ to an integrable function `g` which is measurable with
+  respect to `ℱ∞ := ⨆ n, ℱ n` and
+(b) if `f` is actually a martingale, `f n = 𝔼[g | ℱ n]` almost everywhere.
+(c) Finally, if `h` is integrable and measurable with respect to `ℱ∞`, `(𝔼[h | ℱ n])ₙ` is a
+  uniformly integrable martingale which converges to `h` almost everywhere and in L¹.
+
+The proof is quite simple. (a) follows directly from the a.e. martingale convergence theorem
+and the Vitali convergence theorem. Mathematically, one first have to observe that uniform
+integrability implies uniform boundedness in L¹. Indeed, if
+$$
+  \lim_{\lambda \to \infty} \sup_{n \ge 0} \mathbb{E}(|f_n|\mathbf{1}_{\{|f_n| > \lambda\}}) = 0,
+$$
+then there exists some $\lambda$ such that
+$\sup_{n \ge 0} \mathbb{E}(|f_n|\mathbf{1}_{\{|f_n| > \lambda\}}) \le 1$. So,
+$$
+  \sup_{n \ge 0} \mathbb{E}|f_n| \le
+    \sup_{n \ge 0} \mathbb{E}|f_n|\mathbf{1}_{\{|f_n| \le \lambda\}} +
+    \sup_{n \ge 0} \mathbb{E}|f_n|\mathbf{1}_{\{|f_n| > \lambda\}} \le
+    \lambda \mu(\Omega) + 1 < \infty.
+$$
+However, by the very definition we used for uniform integrability in the probability sense,
+uniform integrability in Lean directly requires L¹ boundedness and so the above is unnecessary.
```

Comment:
That paragraph was not clear to me: at the end I was trying to figure out why our definition of uniform integrable was not the "mathematical" one, and whether it was too strong and we could prove results for something weaker.
The answer is of course the result `measure_theory.uniform_integrable_iff`: the two definitions are equivalent.

I suggest you remove that paragraph and write only the proof for our definition. You could also add a note around line 271 to refer the reader to our definition and to the theorem `uniform_integrable_iff`.